graylog2-server/src/main/java/org/graylog2/shared/bindings/providers/OkHttpClientProvider.java

```diff
@@ -97,8 +105,36 @@ public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
             };
 
             clientBuilder.proxySelector(proxySelector);
+
+            if (!isNullOrEmpty(httpProxyUri.getUserInfo())) {
+                final List<String> list = Splitter.on(":")
+                        .limit(2)
+                        .omitEmptyStrings()
+                        .splitToList(httpProxyUri.getUserInfo());
+                if (list.size() == 2) {
+                    clientBuilder.proxyAuthenticator(new ProxyAuthenticator(list.get(0), list.get(1)));
+                }
+            }
         }
 
         return clientBuilder.build();
     }
+
+    public static class ProxyAuthenticator implements Authenticator {
+        private static final String PROXY_AUTHORIZATION = "Proxy-Authorization";
+        private final String credentials;
+
+        ProxyAuthenticator(String user, String password) {
+            this.credentials = Credentials.basic(requireNonNull(user), requireNonNull(password));
+        }
+
+        @Nullable
+        @Override
+        public Request authenticate(@Nonnull Route route, @Nonnull Response response) throws IOException {
+            if (response.request().header(PROXY_AUTHORIZATION) != null) {
+                return null; // Give up, we've already failed to authenticate.
+            }
+            return response.request().newBuilder().addHeader(PROXY_AUTHORIZATION, credentials).build();
```

Comment:
We're using basic authentication here, no matter what mechanism the `Proxy-Authentication` request header mentions.

We should at least issue a warning if the proxy requests an authorization mechanism which we don't support (which is everything except "Basic") and fall back to no credentials.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Proxy-Authorization
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Proxy-Authenticate

Reply:
Good catch! :+1: